src/components/MessageBot/MessageBot.tsx

```diff
@@ -18,25 +18,41 @@ const Message: StyledComponent<{}, {}, TockTheme> = styled.div`
   white-space: pre-line;
   border-radius: ${props =>
     (props.theme &&
-      props.theme.borderRadius &&
-      `${props.theme.borderRadius} ${props.theme.borderRadius} ${props.theme.borderRadius} 0`) ||
+        props.theme.borderRadius &&
+        `${props.theme.borderRadius} ${props.theme.borderRadius} ${props.theme.borderRadius} 0`) ||
     '1em'};
 
   ${props => (props.theme && props.theme.styles && props.theme.styles.messageBot) || ''}
 `;
 
-const MessageBot: ({ children }: { children: ReactNode }) => JSX.Element = ({
-  children,
-}: {
-  children: ReactNode;
-}) => {
-  return (
-    <MessageContainer>
-      <Message>
-        <span dangerouslySetInnerHTML={{ __html: children ? children.toString() : '' }}></span>
-      </Message>
-    </MessageContainer>
-  );
+const LINK_REGEX: RegExp = /((<a.*?href=["'])?(?:(?:https?):\/\/|www\.)?([a-zA-Z0-9]*\.([a-zA-Z0-9]*\.)?)([a-zA-Z]+)([-A-Z0-9+&@#\/%=~_|$?!:,.]+)?(["'].*?(<?\/a?>))?)/igm;
+const HTML_LINK_REGEX: RegExp = /(<a.*?href=["'].*?["'].*?<?\/a?>)/igm;
+
+const MessageBot: ({ children }: { children: ReactNode }) => JSX.Element = ({ children }: { children: ReactNode }) => {
+
+    function getHtmlContent() {
+        let message = children ? children.toString() : '';
+        if (LINK_REGEX.test(message)) {
+            return message.replace(LINK_REGEX, (replaced) => {
+                if (HTML_LINK_REGEX.test(replaced)) {
+                    return replaced;
+                }
+                if (replaced.indexOf('http') < 0) {
+                    return '<a href="http://' + replaced + '" target="_blank">' + replaced + '</a>';
+                }
+                return '<a href="' + replaced + '" target="_blank">' + replaced + '</a>';
+            });
+        }
+        return message;
+    }
+
+    return (
+        <MessageContainer>
+            <Message>
+                <span dangerouslySetInnerHTML={{__html: getHtmlContent()}}/>
```

Comment:
If bot message contains "`<p>message</p>`" generating HTML becomes "`<span><p>message</p></span>`".

And element p not allowed as child of element span in this context.

Replacing span with a div element?

Reply:
done! 👍